### PR TITLE
Move unboxed array tests under test-core flag

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/1790

File: test/Streamly/Test/Data/Array/Unboxed.hs [hunk 1]
```diff
@@ -8,7 +8,10 @@
 
 module Streamly.Test.Data.Array.Unboxed (main) where
 
-#include "Streamly/Test/Data/Array/CommonImports.hs"
+import Streamly.Data.Fold (Fold)
+import Streamly.Internal.Data.Unboxed (sizeOf)
+import Streamly.Internal.Data.Stream (Stream)
+import Streamly.Test.Common (listEquals)
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Array/Unboxed.hs:14]: Add these after line 20, deduplicate the Data.Unboxed import.

File: test/Streamly/Test/Data/Array/Unboxed.hs [hunk 2]
```diff
@@ -18,10 +21,16 @@ import Streamly.Internal.Data.Unboxed (Unboxed)
 import Test.QuickCheck (chooseInt, listOf)
 import GHC.Ptr (plusPtr)
 
+import qualified Streamly.Data.Fold as Fold
 import qualified Streamly.Internal.Data.Array.Unboxed as A
 import qualified Streamly.Internal.Data.Array.Unboxed.Type as A
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Array/Unboxed.hs:26]: Do we need this? Array.Unboxed should have all we need.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: test/Streamly/Test/Data/Array/Unboxed.hs (hunk 2 above)
Comment (rnjtranjan) [reply test/Streamly/Test/Data/Array/Unboxed.hs:23]: Functions from both modules are used.

**Pull request merged**